docs/test-suite/writing-portable-tests.md

```diff
@@ -0,0 +1,526 @@
+
+An example of such a collective operation is the [MPI_REDUCE](https://www.mpi-forum.org/docs/mpi-4.1/mpi41-report/node130.htm#Node130) call. It reduces data elements from multiple processes with a certain operation, e.g. it takes the sum of all elements or multiplication of all elements.
+
+#### The mpi4py test
+In this example, we will implement a test that does an `MPI_Reduce` on the rank, using the sum operation. This makes it easy to validate the result, as we know that for N processes, the theoretical sum of all ranks (0, 1, ... N-1) is `(N * (N-1) / 2)`.
+
+Our initial code is a python script `mpi4py_reduce.py`, which can be found in `tutorials/mpi4py/src/mpi4py_reduce.py` in the [EESSI test suite](https://github.com/EESSI/test-suite/) repository:
+```python
+#!/usr/bin/env python
+"""
+MPI_Reduce on MPI rank. This should result in a total of (size * (size - 1) / 2),
+where size is the total number of ranks.
+Prints the total number of ranks, the sum of all ranks, and the time elapsed for the reduction."
+"""
+
+import argparse
+import time
+
+from mpi4py import MPI
+
+parser = argparse.ArgumentParser(description='mpi4py reduction benchmark',
+                                 formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument('--n_warmup', type=int, default=100,
+                    help='Number of warmup iterations')
+parser.add_argument('--n_iter', type=int, default=1000,
+                    help='Number of benchmark iterations')
+args = parser.parse_args()
+
+n_warmup = args.n_warmup
+n_iter = args.n_iter
+
+size = MPI.COMM_WORLD.Get_size()
+rank = MPI.COMM_WORLD.Get_rank()
+name = MPI.Get_processor_name()
+
+# Warmup
+t0 = time.time()
+for i in range(n_warmup):
+    total = MPI.COMM_WORLD.reduce(rank)
+
+# Actual reduction, multiple iterations for accuracy of timing
+t1 = time.time()
+for i in range(n_iter):
+    total = MPI.COMM_WORLD.reduce(rank)
+t2 = time.time()
+total_time = (t2 - t1) / n_iter
+
+if rank == 0:
+    print(f"Total ranks: {size}")
+    print(f"Sum of all ranks: {total}")  # Should be (size * (size-1) / 2)
+    print(f"Time elapsed: {total_time:.3e}")
+```
+
+Assuming we have `mpi4py` available, we could run this manually using
+```
+$ mpirun -np 4 python3 mpi4py_reduce.py
+Total ranks: 4
+Sum of all ranks: 6
+Time elapsed: 3.609e-06
+```
+
+This started 4 processes, with ranks 0, 1, 2, 3, and then summed all the ranks (`0+1+2+3=6`) on the process with rank 0, which finally printed all this output. The whole reduction operation is performed `n_iter` times, so that we get a more accurate timing.
```

Comment:
```suggestion
This started 4 processes, with ranks 0, 1, 2, 3, and then summed all the ranks (`0+1+2+3=6`) on the process with rank 0, which finally printed all this output. The whole reduction operation is performed `n_iter` times, so that we get a more reproducible timing.
```